fitbenchmarking/utils/options.py

```diff
@@ -227,79 +234,115 @@ def __init__(self, file_name=None, results_directory: str = ""):
-        self.colour_map = self.read_value(plotting.getstr, 'colour_map')
-        self.colour_ulim = self.read_value(plotting.getfloat, 'colour_ulim')
-        self.cmap_range = self.read_value(plotting.getrng, 'cmap_range')
+
+        if 'make_plots' in additional_options:
+            self.make_plots = additional_options['make_plots']
+        else:
+            self.make_plots = self.read_value(
+                plotting.getboolean, 'make_plots', additional_options)
+
+        self.colour_map = self.read_value(
+            plotting.getstr, 'colour_map', additional_options)
+        self.colour_ulim = self.read_value(
+            plotting.getfloat, 'colour_ulim', additional_options)
+        self.cmap_range = self.read_value(
+            plotting.getrng, 'cmap_range', additional_options)
 
         self.comparison_mode = self.read_value(plotting.getstr,
-                                               'comparison_mode')
-        self.table_type = self.read_value(plotting.getlist, 'table_type')
+                                               'comparison_mode',
+                                               additional_options)
+
+        self.table_type = self.read_value(plotting.getlist, 'table_type',
+                                          additional_options)
 
         output = config['OUTPUT']
-        if results_directory == "":
-            self.results_dir = self.read_value(output.getstr, 'results_dir')
-        else:
-            self.results_dir = results_directory
+
+        self.results_dir = self.read_value(output.getstr, 'results_dir',
+                                           additional_options)
 
         logging = config['LOGGING']
-        self.log_append = self.read_value(logging.getboolean, 'append')
-        self.log_file = self.read_value(logging.getstr, 'file_name')
-        self.log_level = self.read_value(logging.getstr, 'level')
+
+        self.log_append = self.read_value(logging.getboolean, 'append',
+                                          additional_options)
+
+        self.log_file = self.read_value(logging.getstr, 'file_name',
+                                        additional_options)
+
+        self.log_level = self.read_value(logging.getstr, 'level',
+                                         additional_options)
 
         self.external_output = self.read_value(logging.getstr,
-                                               'external_output')
+                                               'external_output',
+                                               additional_options)
 
         if self.error_message != []:
             raise OptionsError('\n'.join(self.error_message))
 
-    def reset(self):
-        """
-        Resets options object when running multiple problem groups.
-        """
-        self.__init__(self.stored_file_name, self.results_dir)
-
-    def read_value(self, func, option):
+    def read_value(self, func, option, additional_options_func):
```

Comment:
Did you intend to leave this as "additional_options_func"? I'm not sure the "_func" adds anything.

Reply:
I got rid of the `func` and then added it back when it wasn't working, so I'll remove it again now